hail/src/main/scala/is/hail/expr/ir/BaseIR.scala

```diff
@@ -6,6 +6,10 @@ import is.hail.types.virtual.Type
 import is.hail.utils._
 import is.hail.utils.StackSafe._
 
+case class Name(str: String) extends AnyVal {
+  override def toString: String = str
+}
```

Comment:
Think you can do this in one line :)
```suggestion
case class Name(override val toString: String) extends AnyVal
```

Reply:
I like your idea of using object equality rather than string equality.
Keeping the string, however, might allow us to better keep track of where a name came from through lowering.

Reply:
Absolutely! From the description:
> keep a string in the Name, but no longer require it to be unique. Instead it's just a suggestion for how to show the name in printouts, adding a uniqueifying suffix as needed. With NormalizeNames gone, this would let us preserve meaningful variable names further in the lowering pipeline.

Reply:
> Think you can do this in one line :)

I could... but I'd rather not commit to this choice of `toString` permanently.